kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/workloadrunners/WriteRunner.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.benchmarks.workloadrunners;
+
+import static io.delta.kernel.internal.util.Utils.toCloseableIterator;
+
+import io.delta.kernel.*;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.defaults.benchmarks.models.WorkloadSpec;
+import io.delta.kernel.defaults.benchmarks.models.WriteSpec;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.transaction.UpdateTableTransactionBuilder;
+import io.delta.kernel.utils.CloseableIterable;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.DataFileStatus;
+import io.delta.kernel.utils.FileStatus;
+import java.io.IOException;
+import java.util.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+/**
+ * A WorkloadRunner that executes write workloads as benchmarks. This runner performs one or more
+ * commits to a Delta table and measures the performance of those commits.
+ *
+ * <p>The runner executes commits specified in the {@link WriteSpec}, where each commit contains a
+ * set of Delta log actions (add/remove files) defined in external JSON files.
+ *
+ * <p>If run as a benchmark using {@link #executeAsBenchmark(Blackhole)}, this measures the time to
+ * execute all commits. Setup (loading commit files) and cleanup (reverting changes) are not
+ * included in the benchmark timing.
+ */
+public class WriteRunner extends WorkloadRunner {
+  private final Engine engine;
+  private final WriteSpec workloadSpec;
+  private final List<List<DataFileStatus>> commitContents;
+  private Snapshot currentSnapshot;
+  private Set<String> initialDeltaLogFiles;
+
+  /**
+   * Constructs the WriteRunner from the workload spec and engine.
+   *
+   * @param workloadSpec The write workload specification.
+   * @param engine The engine to use for executing the workload.
+   */
+  public WriteRunner(WriteSpec workloadSpec, Engine engine) {
+    this.workloadSpec = workloadSpec;
+    this.engine = engine;
+    this.commitContents = new ArrayList<>();
+  }
+
+  @Override
+  public void setup() throws Exception {
+    String tableRoot = workloadSpec.getTableInfo().getResolvedTableRoot();
```

Comment:
Do we consider copying this table to a temp location so that two runner could test it without affect each other?